client/src/containers/Course/CourseDetail.js

```diff
@@ -14,11 +14,8 @@ function CourseDetail(props) {
     let distinctInstructors = [
       ...new Set(course.sections.filter((x) => x.instructor).map((x) => x.instructor)),
     ];
-    distinctInstructors.forEach((x) => {
-      if (!(x in props.ratingsMap)) {
-        props.getProfRating(x);
-      }
-    });
+    props.getBatchProfRating(distinctInstructors
```

Comment:
maybe move this filtering and mapping logic to a const? like 
const currentInstructors = ...